app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -230,7 +230,20 @@ public function fetchTransactionInfo(Mage_Payment_Model_Info $payment, $transact
     public function isAvailable($quote = null)
     {
         if(!empty($quote)) {
-            return $this->boltHelper()->canUseBolt($quote);
+            $canUseBolt = $this->boltHelper()->canUseBolt($quote);
+            $hidePayment = false;
+
+            $routeName= Mage::app()->getRequest()->getRouteName();
+            $moduleName = Mage::app()->getRequest()->getControllerModule();
+            $controllerName = Mage::app()->getRequest()->getControllerName();
+            if (
+                ($routeName === 'onestepcheckout' && $moduleName === 'Idev_OneStepCheckout')
```

Comment:
@YevhenShBolt 

Do you think that we can do something like 
```
$routeName= Mage::app()->getRequest()->getRouteName();
$controllerName = Mage::app()->getRequest()->getControllerName();

$onepageCheckoutUrl = Mage::app()->getUrl('checkout/onepage');
$requestedUrl = $routeName . '/' . $controllerName;

if ($onepageCheckoutUrl === $requestedUrl) {
     $hidePayment = Mage::getStoreConfigFlag('payment/boltpay/hide_on_checkout');
}
```

This may be a way to generically support all one page checkout extensions that override the standard `checkout/onepage`.

By the way, who uses Idev_OneStepCheckout?

Reply:
Yevhen created this fix for a live issue with Tonerprice

Reply:
@matt-thomason That makes sense, but it seems like this should be in there custom code as opposed to the plugin which could take the opportunity to solve this in a more robust and universal way.

Reply:
refactored.